pkg/ovs/ovsctl/ofctl_test.go

```diff
@@ -0,0 +1,237 @@
+	out, err := client.DumpMatchedFlow(matchFlow)
+	assert.NoError(t, err)
+	expectedFlow := "table=2, priority=0, n_packets=0, n_bytes=0, idle_age=65534, hard_age=65534, actions=drop "
+	assert.Equal(t, expectedFlow, out)
+}
+
+func TestDumpTableFlows(t *testing.T) {
+	client := &ovsCtlClient{
+		bridge: "br-int",
+	}
+	runCommand = func(cmd string) ([]byte, error) {
+		var x = []byte{}
+		table := cmd[strings.Index(cmd, "table="):strings.Index(cmd, " --names")]
+		for i := 0; i < len(testDumpFlows); i++ {
+			if !strings.Contains(testDumpFlows[i], table) {
+				continue
+			}
+			b := []byte(testDumpFlows[i])
+			for j := 0; j < len(b); j++ {
+				x = append(x, b[j])
+			}
+			x = append(x, "\n"...)
+		}
+		return x, nil
+	}
+	defer func() {
+		runCommand = runCmd
+	}()
+	out, err := client.DumpTableFlows(3)
+	assert.NoError(t, err)
+	expectedFlows := []string{"table=3, priority=200, n_packets=29233419, n_bytes=2703471860, idle_age=8, hard_age=65534,in_port=2 actions=load:0x2->NXM_NX_REG0[0..3],resubmit(,4) ", "table=3, priority=200, n_packets=0, n_bytes=0, idle_age=65534, hard_age=65534,in_port=1 actions=load:0x1-> "}
+	assert.Equal(t, expectedFlows, out)
+}
+
+func TestDumpGroups(t *testing.T) {
+	client := &ovsCtlClient{
+		bridge: "br-int",
+	}
+	runCommand = func(cmd string) ([]byte, error) {
+		var x = []byte{}
+		for i := 0; i < len(testDumpGroups); i++ {
+			b := []byte(testDumpGroups[i])
+			for j := 0; j < len(b); j++ {
+				x = append(x, b[j])
+			}
+			x = append(x, "\n"...)
+		}
+		return x, nil
+	}
+	defer func() {
+		runCommand = runCmd
+	}()
+	out, err := client.DumpGroups()
+	assert.NoError(t, err)
+	expectedGroups := []string{"group_id=1,type=all,bucket=mod_dl_src=00:00:00:11:11:11,mod_dl_dst=00:00:00:22:22:22,output:2,bucket=mod_dl_src=00:00:00:11:11:11,mod_dl_dst=00:00:00:22:22:22,output:3", "group_id=2,type=indirect,bucket=mod_dl_src=00:00:00:99:11:11,mod_dl_dst=00:00:00:99:22:22,output:2", "group_id=3,type=select,bucket=bucket=output:1,output:2,bucket=output:3,bucket=output:4", "group_id=4,type=ff,bucket=watch_port:2,watch_group:2,output:3,watch_port:3,watch_group:3,output:4"}
+	assert.Equal(t, expectedGroups, out)
+}
+
+func TestDumpGroup(t *testing.T) {
+	client := &ovsCtlClient{
+		bridge: "br-int",
+	}
+	groupID := 3
+	runCommand = func(cmd string) ([]byte, error) {
+		var x = []byte{}
+		id := cmd[strings.Index(cmd, "br-int")+7 : strings.Index(cmd, " -O")]
+		for i := 0; i < len(testDumpGroups); i++ {
+			if !strings.Contains(testDumpGroups[i], id) {
+				continue
+			}
+			b := []byte(testDumpGroups[i])
+
+			for j := 0; j < len(b); j++ {
+				x = append(x, b[j])
+			}
+			x = append(x, "\n"...)
+
+		}
+		return x, nil
+	}
+	defer func() {
+		runCommand = runCmd
+	}()
+	out, err := client.DumpGroup(uint32(groupID))
+	assert.NoError(t, err)
+	expectedGroup := "group_id=3,type=select,bucket=bucket=output:1,output:2,bucket=output:3,bucket=output:4"
+	assert.Equal(t, expectedGroup, out)
```

Comment:
during assertequal improve the output readability like add function name when equal fails